packages/@mantine/core/src/components/Tree/use-tree.ts

```diff
@@ -20,20 +20,29 @@ function getInitialState(
     acc[node.value] = node.value in initialState ? initialState[node.value] : node.value === value;
 
     if (Array.isArray(node.children)) {
-      getInitialState(initialState, node.children, value, acc);
+      getInitialExpandedState(initialState, node.children, value, acc);
     }
   });
 
   return acc;
 }
 
+function getInitialCheckedState(initialState: string[], data: TreeNodeData[], acc: string[] = []) {
```

Comment:
If acc is not part of recursive function, it should be declared in the function body.